cairo/crates/pragma_dispatcher/src/routers/feed_types/contracts/twap.cairo

```diff
@@ -0,0 +1,187 @@
+                .call_calculate_twap(data_type, aggregation_mode, start_timestamp, duration);
+
+            let mut update = BytesTrait::new_empty();
+            update.append_u256(feed.id().into());
+            update.append_u64(get_block_timestamp());
+            update.append_u16(0); // TODO: num sources ?
+            update.append_u8(decimals.try_into().unwrap());
+            update.append_u256(twap_price.into());
+            update.append_u256(0); // TODO: time period ?
+            update.append_u256(0); // TODO: start price ?
+            update.append_u256(0); // TODO: end price ?
+            update.append_u256(0); // TODO: total volume ?
+            update.append_u256(0); // TODO: number of data points ?
+
+            update
+        }
+    }
+
+
+    // ================== PRIVATE IMPLEMENTATIONS ==================
+
+    #[generate_trait]
+    impl InternalImpl of InternalTrait {
+        /// Initializes the contract storage.
+        /// Called only once by the constructor.
+        fn initializer(
+            ref self: ContractState,
+            summary_stats_address: ContractAddress,
+            feed_type_id: FeedTypeId,
+        ) {
+            // [Check] Contracts are not zero
+            assert(!summary_stats_address.is_zero(), errors::SUMMARY_STATS_IS_ZERO);
+
+            // [Check] Feed type id is valid
+            let feed_type = match FeedTypeTrait::from_id(feed_type_id) {
+                Result::Ok(f) => f,
+                Result::Err(e) => panic_with_felt252(e.into())
+            };
+            // [Check] Feed type variant & extract parameters
+            let (simple_data_type, aggregation_mode, duration) = match feed_type {
+                FeedType::Twap(variant) => {
+                    match variant {
+                        TwapVariant::SpotMedianOneDay => {
+                            (SimpleDataType::Spot, AggregationMode::Median, Duration::OneDay)
+                        },
+                    }
+                },
+                _ => panic_with_felt252(errors::INVALID_FEED_TYPE_FOR_CONTRACT)
+            };
+
+            // [Effect] Save feed type to storage
+            self.feed_type.write(feed_type);
+            // [Effect] Set parameters depending on the variant
+            self.simple_data_type.write(simple_data_type);
+            self.aggregation_mode.write(aggregation_mode);
+            self.duration.write(duration);
+
+            // [Interaction] Emit new router deployed
+            self
+                .emit(
+                    FeedTypeRouterDeployed {
+                        sender: get_caller_address(),
+                        feed_type_id: feed_type_id,
+                        router_address: get_contract_address(),
+                    }
+                )
+        }
+    }
+
+    impl SummaryStatsWrapper of ISummaryStatsWrapper<ContractState> {
+        /// Calls calculate_volatility from the Summary Stats contract.
+        fn call_calculate_volatility(
```

Comment:
Why is that here ?

Reply:
The interface `ISummaryStatsWrapper` is the same for every feed types so we need to implement every functions. Any better idea?

Reply:
Yes, I think we can just use the raw dispatcher here

Reply:
done